{{ cookiecutter.project_slug }}/{{ cookiecutter.project_slug }}/users/views.py

```diff
@@ -36,8 +37,13 @@ def change_email_request(self, request, *args, **kwargs):
 
         user = request.user
         self.check_object_permissions(request, user)
-
-        user.new_email = serializer.data['email']
+        
+        new_email = serializer.data['email']
+                                    
+        if User.objects.filter(email=new_email).exists():
+            raise ValidationError({'email': ['user with this email already exists.']})
```

Comment:
Instead of doing the check inline here. Check out `ChangeEmailRequestSerializer`, we can actually do custom validation there in a more idiomatic way (since validation typically happens at the serializer level)

I would suggest modifying the `ChangeEmailRequestSerializer` to something like this.

```typescript
class ChangeEmailRequestSerializer(serializers.Serializer):
    def validate_email(self, value):
        if User.objects.filter(email=value).exists():
            raise serializers.ValidationError('A user with this email already exists.')
        return value

    email = serializers.EmailField()
```

Check out the docs regarding serializers and field level validation here: https://www.django-rest-framework.org/api-guide/serializers/#field-level-validation

Reply:
The Idea is basically the same we are just moving to a more appropriate place.

Reply:
Okay. Will do.